appsec/src/helper/client.cpp

```diff
@@ -211,152 +211,112 @@ bool client::handle_command(const network::client_init::request &command)
     return !has_errors;
 }
 
-bool client::handle_command(network::request_init::request &command)
+template <typename T> bool client::service_guard()
 {
     if (!service_) {
         // This implies a failed client_init, we can't continue.
-        SPDLOG_DEBUG("no service available on request_init");
+        SPDLOG_DEBUG("no service available on {}", T::name);
         send_error_response(*broker_);
         return false;
     }
 
-    if (!compute_client_status()) {
-        auto response_cf =
-            std::make_shared<network::config_features::response>();
-        response_cf->enabled = false;
-
-        SPDLOG_DEBUG("sending config_features to request_init");
-        try {
-            return broker_->send(response_cf);
-        } catch (std::exception &e) {
-            SPDLOG_ERROR(e.what());
-        }
-
-        return true;
-    }
-
-    // During request init we initialize the engine context
-    context_.emplace(*service_->get_engine());
+    return true;
+}
 
-    SPDLOG_DEBUG("received command request_init");
+template <typename T>
+std::shared_ptr<typename T::response> client::publish(
+    typename T::request &command)
+{
+    SPDLOG_DEBUG("received command {}", T::name);
 
-    auto response = std::make_shared<network::request_init::response>();
+    auto response = std::make_shared<typename T::response>();
     try {
+        // NOLINTNEXTLINE(bugprone-unchecked-optional-access)
         auto res = context_->publish(std::move(command.data));
         if (res) {
-            switch (res->type) {
-            case engine::action_type::block:
-                response->verdict = network::verdict::block;
-                response->parameters = std::move(res->parameters);
-                break;
-            case engine::action_type::redirect:
-                response->verdict = network::verdict::redirect;
-                response->parameters = std::move(res->parameters);
-                break;
-            case engine::action_type::record:
-            default:
-                response->verdict = network::verdict::record;
-                response->parameters = {};
-                break;
+            for (auto &act : res->actions) {
```

Comment:
From the WAF to the message, it seems we're converting the actions three times, it's not very efficient. Perhaps you could remove the conversion between the subscriber and the engine?